backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -690,6 +690,33 @@ class VeloxSparkPlanExecApi extends SparkPlanExecApi {
     GenericExpressionTransformer(substraitExprName, children, expr)
   }
 
+  /** Generate an expression transformer to transform JsonToStructs to Substrait. */
+  override def genFromJsonTransformer(
+      substraitExprName: String,
+      children: Seq[ExpressionTransformer],
+      expr: JsonToStructs): ExpressionTransformer = {
+    if (!GlutenConfig.getConf.enableVeloxFromJsonFunction) {
+      throw new GlutenNotSupportException("'from_json' is not supported in Velox")
+    }
+    if (!SparkShimLoader.getSparkShims.fromJsonSupportPartialResults) {
```

Comment:
So will it not be offloaded to velox for spark version < 3.4 ? Are there some cases where we can still offload if this is false for lower versions?

Reply:
> So will it not be offloaded to velox for spark version < 3.4 ?

Yes, as in Velox we only implement the code path of "enablePartialResults" now.

> Are there some cases where we can still offload if this is false for lower versions?

You can change `fromJsonSupportPartialResults` to true and run test suites "GlutenJsonExpressionsSuite" and "GlutenJsonFunctionsSuite".